syp.biz/SockJS.NET/syp.biz.SockJS.NET.Client/Transports/Lib/Driver/WebSocketDriver.cs

```diff
@@ -69,6 +69,11 @@ private async void ReceiveLoop(object obj)
                     var buffer = new byte[1024];
                     var segment = new ArraySegment<byte>(buffer);
                     var result = await this._socket.ReceiveAsync(segment, this._cancel.Token);
+                    if (result.MessageType == WebSocketMessageType.Close)
```

Comment:
can you change to call `Close(WebSocketCloseStatus status, string reason)` and handle it there instead?
Since this is not an implementation based directly on the JavaScript SockJS implementation we have a bit of a free rein here and probably should do it a bit more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Reply:
Got it, but `Close(WebSocketCloseStatus status, string reason)` will have to be changed also, because for now, it doesn't notify at the `SockJs` event listener level about close event.
In other words, if not changing `Close(WebSocketCloseStatus, string)` method, this delegate wont be called.
<pre>
SockJs.AddEventListener("close", async (sender, e) =>
                {
                    // event not fired
                });
</pre>